ampd/src/solana/msg_verifier.rs

```diff
@@ -119,7 +116,12 @@ fn find_first_log_message_match(
     for (i, log) in log_messages.iter().enumerate() {
         match GatewayEvent::parse_log(log) {
             Some(parsed_ev) => {
-                let verified = parsed_ev == message
+                let arch_gw_event = parsed_ev.parse();
+
+                println!("{:?}<<<______parsed__________", arch_gw_event);
+                println!("{:?}<<<______parsed_______msg___", message);
```

Comment:
I am still debugging a comparison error. Seems has to do with the `sender` field of the message.

Reply:
But i think this might have further implications, let's analyse it together:

The type we are using for parsing the Axelar side event is [TxEventConfirmation](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/contracts/voting-verifier/src/events.rs#L193):

```rust
#[cw_serde]
pub struct TxEventConfirmation {
    pub tx_id: nonempty::String,
    pub event_index: u32,
    pub destination_address: Address,
    pub destination_chain: ChainName,

    pub source_address: Address, //  <---------------------------------- THE LATER SENDER FIELD IN COMPARISON.


    /// for better user experience, the payload hash gets encoded into hex at the edges (input/output),
    /// but internally, we treat it as raw bytes to enforce its format.
    #[serde(with = "axelar_wasm_std::hex")]
    #[schemars(with = "String")] // necessary attribute in conjunction with #[serde(with ...)]
    pub payload_hash: [u8; 32],
}
```

Such [Address](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/packages/router-api/src/primitives.rs#L79-L114) type  expects to parse a String directly.

In our tests, we use that structures to [simulate an incoming event from the axelar chain](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/ampd/src/handlers/solana_verify_msg.rs#L260-L267) and then [we pass it to the handlers](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/ampd/src/handlers/solana_verify_msg.rs#L277-L278) encoded.

Then in handlers, we have defined [our own structures](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/ampd/src/handlers/solana_verify_msg.rs#L29-L52) for [decoding](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/ampd/src/handlers/solana_verify_msg.rs#L135-L150) the data we need. As a subtype of that structure, we can find the `Message` with the source address, which needs to be a string or it will fail deserializing from the Axelar event.

Thus i think we are forced here to parse the bytes from the tx log line into a Pubkey, then convert that into it's String representation (on my mind the base58 encoded pubkey) in order to be able to properly compare with the Axelar side, which i think it also provides us base58 solana encoded address. [Thus the comparison logic](https://github.com/eigerco/axelar-amplifier/blob/b0deb64d369eba32abd264c03bff397845398894/ampd/src/solana/msg_verifier.rs#L12-L31) would end like this:

```rust
impl PartialEq<&Message> for &ArchivedGatewayEvent {
    fn eq(&self, msg: &&Message) -> bool {
        match self {
            ArchivedGatewayEvent::CallContract(ArchivedCallContract {
                sender,
                destination_chain,
                destination_address,
                payload: _,
                payload_hash,
            }) => {
                let sender = Pubkey::from(*sender).to_string();  // <---- Conversion here, in order to properly compare string representations.
                    sender == msg.source_address
                    && msg.destination_chain == destination_chain.as_str()
                    && msg.destination_address == destination_address.as_str()
                    && *payload_hash == msg.payload_hash
            }
            _ => false,
        }
    }
}
``` 
With the above changes, the tests are passing now. Let me know you thoughts.

Reply:
> that into it's String representation (on my mind the base58 encoded pubkey) 
Yes, I think we can assume that all Pubkey string representations are going to be the base58 encoded version.

Regarding the comparison--
I'd try it the other way around: `sender == Pubkey::from_str(msg.source_address.as_srt())`
Allocating a `[u8; 32]` (internal repr of a Pukbkey) on the stack is cheaper than allocating & encoding a string on the heap.

But generally, the idea is correct 👍

Reply:
fixed at 6db9a2fd76922306b81c8b43d49fc143fb15b367